src/functions/secrets.ts

```diff
@@ -0,0 +1,104 @@
+import { createSecret, getSecret, patchSecret, Secret } from "../gcp/secretManager";
+import { Options } from "../options";
+import { FirebaseError } from "../error";
+import { logWarning } from "../utils";
+import { promptOnce } from "../prompt";
+import { validateKey } from "./env";
+
+const FIREBASE_MANGED = "firebase-managed";
+
+/**
+ * Returns true if secret is managed by Firebase.
+ */
+function isFirebaseManaged(secret: Secret): boolean {
+  return Object.keys(secret.labels || []).includes(FIREBASE_MANGED);
+}
+
+/**
+ * Return labels to mark secret as managed by Firebase.
+ * @internal
+ */
+export function labels(): Record<string, string> {
+  return { [FIREBASE_MANGED]: "true" };
+}
+
+function transformKey(key: string): string {
```

Comment:
I might rename this more descriptively like `toUpperSnakeCase`

Reply:
Huh good point. Since you aren't setting more than one secrets at a time though, I don't think collisions are particularly dangerous here. If we make any sort of transformation, we ask for user confirmation anyway.